sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/SaplingParamTool.kt

```diff
@@ -119,44 +247,69 @@ object SaplingParamTool {
                     throw TransactionEncoderException.FetchParamsException(it)
                 }
             }.onSuccess {
-                twig("Fetch and write of ${paramsToFetch.fileName} succeeded.")
-            }
-        }
-    }
+                twig(
+                    "Fetch and write of the temporary ${temporaryFile.name} succeeded. Moving it to the final " +
+                        "destination"
+                )
+                val resultFile = File(paramsToFetch.destinationDirectory, paramsToFetch.fileName)
+                if (!renameParametersFile(temporaryFile, resultFile) ||
+                    !isFileHashValid(resultFile, paramsToFetch.fileHash)
```

Comment:
I feel like we should validate the hash before the rename, so that once the file is in place it is guaranteed to be known good.

Reply:
Yes, agree. I wanted to validate also the rename operation, but it's probably unnecessary worry, which brings also a pitfall of needed deletion of the copied, but not validated file. Changed.